cppcache/include/geode/DataInput.hpp

```diff
@@ -502,14 +501,14 @@ class APACHE_GEODE_EXPORT DataInput {
   const uint8_t* m_buf;
   const uint8_t* m_bufHead;
   size_t m_bufLength;
-  std::reference_wrapper<const std::string> m_poolName;
+  const std::string& m_poolName;
```

Comment:
In stead of keeping a ref to pool name can we keep a ref to the pool itself? To construct we look up the pool to get the name and then when we need a pool again we look up the pool using the name. Seems inefficient when we can just ref the pool.